### PR TITLE
Playwright Tests: Extract Search Query Response to Utils

### PR DESCRIPTION
**Pull request by ardelato**
## Description

This came up in https://github.com/iFixit/react-commerce/pull/1180#discussion_r1055888310.
We have this piece of code duplicated across some of our current tests, let's just extract it to a utils file and reuse it.

While extracting it, I also simplified the matching logic a bit.

## QA 

- [ ] CI should pass

Connects: https://github.com/iFixit/react-commerce/issues/1111

### CODE REVIEW

**Comment by jordycosta** [comment]
QA 🟢 

CI is passing 
[View `E2E tests / playwright-run` report](https://github.com/iFixit/react-commerce/actions/runs/3767362736/jobs/6404817087)

Screenshot 2022-12-26 105304

File: frontend/tests/playwright/utils.ts [hunk 1]
```diff
@@ -22,6 +22,15 @@ export const interceptLogin = async (page: Page, userParams?: Object) => {
    );
 };
 
+export const waitForAlgoliaSearch = async (page: Page) => {
+   return page.waitForResponse(async (response) => {
+      return (
+         response.url().includes('1/indexes/*/queries') &&
```
Comment (mlahargou) [inline frontend/tests/playwright/utils.ts:28]: Not super familiar with the algolia requests, but why is there a `1` at the begining? Also, we did we drop the `algolia` check?

**Review by mlahargou — APPROVED** [review]
CR 📱

File: frontend/tests/playwright/utils.ts (hunk 1 above)
Comment (ardelato) [reply frontend/tests/playwright/utils.ts:28]: I am not sure about the `1` either. This is similar to what we do with Cypress 
https://github.com/iFixit/react-commerce/blob/main/frontend/tests/cypress/support/commands.ts#L28

**Pull request merged**